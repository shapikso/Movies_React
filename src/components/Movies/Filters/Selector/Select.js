import React from 'react';
import SelectItem from "./SelectItem";

const Select = ({options}) => {
    return (
        <select name="movie-status" className="basic-field">
            {options.map((el, index) => <SelectItem key={index} value={el[0]} title={el[1]} />)}

        </select>
    );
};

export default Select;