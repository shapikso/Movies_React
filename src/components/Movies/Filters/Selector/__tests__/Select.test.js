import React from 'react';
import {shallow, mount} from 'enzyme';
import Select from "../Select";

describe('Select', ()=> {
    const props = {
        options: [['en','English'],['jp','Japanese']],
        value: 'en',
        setSelector: jest.fn()
    };
    it('should render correctly', () => {
        const component = shallow(<Select {...props} />);
        expect(component).toMatchSnapshot();
    });
    it('should checked render value', () => {
        const component = mount(<Select {...props} />);
        expect(component.find('select').getElement().props.value).toEqual(props.value);
    });
    it('should checked render all options', () => {
        const component = mount(<Select {...props} />);
        expect(component.find('select').getElement().props.children).toHaveLength(props.options.length);
    });
    it('should change value', () => {
        const component = mount(<Select {...props} />);
        component.find('select').find('select').simulate('change', {target: { value : 'jp'}});
        expect(props.setSelector).toHaveBeenCalledWith('jp');
    });
});