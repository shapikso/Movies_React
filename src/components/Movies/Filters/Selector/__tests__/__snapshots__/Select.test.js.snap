// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should render correctly 1`] = `ShallowWrapper {}`;
